daemon/src/capability.rs

```diff
@@ -0,0 +1,39 @@
+pub mod http {
+    use std::error::Error;
+
+    pub enum Site {
+        Site(String),
+    }
```

Comment:
This looks weird to me. I'm having a hard time imagining what other variants this enum could grow, and the fact that the enum name and variant name are the same is actually a clippy lint I think.  I would just use a newtype struct like:

```
pub struct Site(String);
```

This has the nice benefit of being guaranteed by the compiler to be the same size as a plain `String` but with a new type for type checking.

Reply:
This is something I woke up thinking about actually.

Reply:
I'm into it.  I hadn't actually grasped that you were hitting the stackexchange api generally and that the site searched was a parameter and that's what this string represented.  I think looking at it from a maintenance perspective, since more stack exchanges are created all the time the site-name is probably something that should eventually come from the client program so that a user can ask 7roxy to search any stackexchange site without having us having to mention every single SE site by name in the code.